File: app/src/main/java/com/example/architechturestartercode/db/AppDatabase.java
package com.example.architechturestartercode.db;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.example.architechturestartercode.datasource.movies.local.MoviesDao;
import com.example.architechturestartercode.model.Movie;

@Database(entities = {Movie.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {
    public abstract MoviesDao moviesDao();

    private static AppDatabase INSTANCE;

    public static AppDatabase getInstance(Context context) {
        if (INSTANCE == null) {
            INSTANCE = Room.databaseBuilder(context,
                    AppDatabase.class, "moviesDB")
                    .build();
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/example/architechturestartercode/db/AppDataBase.kt
package com.example.architechturestartercode.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase
import com.example.architechturestartercode.data.movie.datasource.local.MoviesDao
import com.example.architechturestartercode.data.movie.model.Movie

@Database(entities = [Movie::class], version = 1)
abstract class AppDataBase : RoomDatabase() {
    abstract fun moviesDao(): MoviesDao?

    companion object {
        private var instance: AppDataBase? = null

        fun getInstance(context: Context): AppDataBase {
            if (instance == null) {
                instance = Room.databaseBuilder(
                    context.getApplicationContext(),
                    AppDataBase::class.java,
                    "moviesdb"
                ).build()
            }
            return instance!!
        }
    }
}
